chatkit-core/src/main/kotlin/com/pusher/chatkit/model/network/Room.kt

```diff
@@ -0,0 +1,65 @@
+package com.pusher.chatkit.model.network
+
+import com.pusher.chatkit.CustomData
+import com.pusher.chatkit.cursors.Cursor
+
+/*
+ * ENTITIES
+ */
+internal data class RoomApiType(
+        val id: String,
+        val createdById: String,
+        val name: String,
+        val pushNotificationTitleOverride: String?,
+        val private: Boolean,
+        val customData: CustomData?,
+        val lastMessageAt: String?,
+        val createdAt: String,
+        val updatedAt: String,
+        val deletedAt: String?
+)
+
+internal data class ReadStateApiType(
+        val roomId: String,
+        val unreadCount: Int,
+        val cursor: Cursor?
+)
+
+internal data class RoomMembershipApiType(val roomId: String, val userIds: List<String>)
+
+/*
+ * SYNC REQUESTS
+ */
+
+internal data class UpdateRoomRequest(
+        val name: String?,
+        val private: Boolean?,
+        val customData: CustomData?
+)
+
+internal data class UpdateRoomRequestWithPNTitleOverride(
```

Comment:
Couldn't we have one class and Gson set up to not serialise nulls? I guess the motivation for this repeated class with such clumsy name is that the backend is not happy receiving null, is that correct?

Reply:
That's existing logic (now those classes just get a better home) and it seems `null` has to be sent so no much value in looking for any improvement here now.